core/src/main/java/org/bitcoinj/core/TransactionOutPoint.java

```diff
@@ -89,19 +90,19 @@ static Connected read(ByteBuffer payload) throws BufferUnderflowException, Proto
 
     class Unconnected implements TransactionOutPoint {
         /** Hash of the transaction to which we refer. */
-        private final Sha256Hash hash;
+        @Nonnull private final Sha256Hash hash;
         /** Which output of that transaction we are talking about. */
         private final long index;
 
-        public Unconnected(Sha256Hash hash, long index) {
+        public Unconnected(@Nonnull Sha256Hash hash, long index) {
```

Comment:
Why are using this `@Nonnull` lately? I thought everything without `@Nullable` is required, at least for our refactored classes?

Reply:
> Why are we using this `@Nonnull` lately?

I guess I"m doing it temporarily as I"m working to document the current behavior and/or start declaring that `null` will no longer be accepted as a parameter.

>  I thought everything without `@Nullable` is required, at least for our refactored classes?

Yes, that is how this class should look when it's finished. Unfortunately nullability is currently hard for me to figure out.